app/Console/Commands/AdserversCrawlCommand.php

```diff
@@ -0,0 +1,180 @@
+
+class AdserversCrawlCommand extends Command
+{
+    protected $broadcast=true;
+    protected $esc;
+    protected $host;
+    protected $registerHostsIfBroadcastedLimit = 3600; // seconds
+    protected $crawlHostsIfLastSeenLimit = 3600*24*14; // seconds
+
+
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'adshares:crawl';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Queries blockchain for available adsevers, downloads available advertisements from each adserver and stores offers in local db. Updates are forwarded to adselect.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->host = config('app.adserver_host');
+
+        // TODO: move to Service
+        // ($walletCommand, $workingDir, $address, $secret, $host, $port)
+        $this->esc = new Esc(
+            config('app.adshares_wallet'),
+            config('app.adshares_workdir'),
+            config('app.adshares_address'),
+            config('app.adshares_secret'),
+            config('app.adshares_node_host'),
+            config('app.adshares_node_port')
+        );
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $this->readBroadcasts();
+        $this->crawlHosts();
+    }
+
+    protected function readBroadcasts()
+    {
+        try {
+            $logMessage = $this->esc->getBroadcastLog(time() - $this->registerHostsIfBroadcastedLimit);
+            print_r($logMessage);
+            $logs = $logMessage->broadcast;
+        } catch (\Exception $e) {
+            $logs = [
+                ['message' => bin2hex($this->host), 'address' => config('app.adshares_address'), 'account_msid' => 1],
+            ];
+            var_dump($logs);
+            // currently it enables to process single own host without blockchain
+//             throw $e;
+        }
+
+        // TODO: check with Jacek - smell logic bugs
+        // TODO: review adseelct json generation filters and double check if this could be copied into adserver database
+
+        if ($logs) {
+            foreach ($logs as $log) {
+                $host = hex2bin($log['message']);
+                if ($this->host == $host) {
+                    // found own broadcast. No need to send it again
+                    $this->broadcast = false;
+                }
+                $this->info("Found $host -> {$log['address']}");
+                // TODO: extract algo
+                if (preg_match('/^([a-z0-9][a-z0-9-]{0,62}\.)+([a-z]{2,})$/i', $host)) {
+                    NetworkHost::registerHost(Esc::normalizeAddress($log['address']), $host);
+                // TODO: check this with Jacek in adserver symfony code
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1135)